feat(msteams): added call option

crisboarna opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->
- Added ability to directly call a user from a found chat with audio / video. (first screencast)
- Added Presence icon for OneOnOne chats (private chats) to display the presence icon for the user based on Microsoft Graph Presence API. (second screencast)

## Screencast
![2024-05-08at08 26 16-ezgif com-resize](https://github.com/raycast/extensions/assets/12763781/0ee3ab45-29e6-4921-8996-5955cbec1806)
![2024-05-10 at 06 27 33](https://github.com/raycast/extensions/assets/12763781/5b7b6c30-14f4-4413-94c3-48f49b011d3e)

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/microsoft-teams/src/actions/callAction.ts:
```diff
@@ -0,0 +1,34 @@
+import { runAppleScript } from "run-applescript";
```
Could we use runAppleScript from `@raycast/utils` instead?

pernielsentikaer reviewed (changes requested):
Thanks for your enhancement @crisboarna 🔥 

@svenwiegand could you also check this?

crisboarna commented:
Hi @pernielsentikaer ,

I have updated it to use the `@raycast\utils` provided utility instead.

pernielsentikaer commented:
I wrote @svenwiegand a message so we can get this merged 🙂

crisboarna commented:
Thank you @pernielsentikaer ! 

I also tried contacting him regarding this PR.

In the meantime I also added Presence icon for `OneOnOne` chats in `Find Chat` (second screencast).

svenwiegand commented:
Thansk for your contribution @crisboarna. I'm currently testing the extension and I'm impressed, with the direct call feature and that it is working based on the Apple Script stuff. Awesome.

But I have issues with the chat status icons: In my test for several users only a progess circle does appear forever:
![CleanShot 2024-05-10 at 17 47 49](https://github.com/raycast/extensions/assets/2285034/4b35f09d-4c01-44bb-b2b1-750fca36233a)

svenwiegand commented:
Besides this, the code looks good to me @crisboarna. Just two tiny things:

- You should update the `CHANGELOG.md` regarding the people state.
- You should update the `README.md` regarding both new features.

crisboarna commented:
Hi @svenwiegand ,

I have not experienced the issue with the presence icon not updating, seems like the API call takes a long time to finish, so I added a timeout with default icon for that scenario.

- Updated `CHANGELOG.md`
- Updated `README.md`
- Added `usePromiseWithTimeout` hook to provide default icon in case of Presence API call taking too long.

Would this PR be ok to merge now ?

svenwiegand commented:
Thanks for the adjustments @crisboarna. From my point of view, this can be merged @pernielsentikaer.
